<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Spatial Analysis of Phenotypic, Genotypic and Environmental Data — EcoGenetics-package • EcoGenetics</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">EcoGenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cran/EcoGenetics">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Spatial Analysis of Phenotypic, Genotypic and Environmental Data</h1>
    </div>

    
    <p>EcoGenetics provides geostatistical tools for the spatial analysis of phenotypic, genotypic and environmental data. The package has been designed to make easy the storing, handling and integration of the available information from multiple sources, under an S4 philosophy.</p>
    

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p></p><table><tr><td>
 Package: </td>
<td> EcoGenetics</td>
</tr><tr><td>
 Type: </td>
<td> Package</td>
</tr><tr><td>
 Version: </td>
<td> 1.2.0-3</td>
</tr><tr><td>
 Date: </td>
<td> 2016-07-15</td>
</tr><tr><td>
 License: </td>
<td> GPL (&gt;=2) </td>
</tr></table>
    <p>----------------------- o-----------------------</p> 
<p><b><em>I. STRUCTURE OF THE PACKAGE</em></b></p>
<p>EcoGenetics has four basic modules. The <strong>base module</strong> is composed by general functions (<a href='eco.lmtree.html'>multiple lm</a>, <a href='eco.detrend.html'>detrending spatial data utility</a>, etc.). 
The <strong>general spatial module</strong> computes global (<a href='eco.gsa.html'>Moran's I</a>, <a href='eco.mantel.html'>Mantel test</a>, etc.) and local (<a href='eco.lsa.html'>Getis-Ord's G*</a>, <a href='eco.lsa.html'>local Moran's I</a>, etc.) spatial tests. These analyses use a <a href='eco.weight.html'>spatial weights matrix</a>, provided by the  <strong>spatial weights module</strong>.  The <strong>lag analysis module</strong> performs two basic analyses: the obtention  of <a href='eco.variogram.html'>variograms</a> and <a href='eco.correlog.html'>correlograms</a> (see also <a href='eco.cormantel.html'>this link</a>). This module uses the other tool provided by the spatial weights module: a sequence of <a href='eco.lagweight.html'>spatial weights matrices</a>.</p>
<p>The package have also special plot methods, as <a href='rankplot-methods.html'>rankplot</a>,  
<a href='forestplot-methods.html'>forestplot</a>, <a href='eco.rasterplot.html'>rasterplot</a> (implemented the last two for 
<a href='eco.lsa.html'>local spatial analysis</a>). Several conversor of data from/to other programs are available, (as to <a href='ecogen2genepop.html'>genepop</a> - an <a href='genepop2ecogen.html'>importer</a> tool is also defined for genepop-, <a href='ecogen2spagedi.html'>SPAGeDi</a>, , etc.). Basic manipulation of genetic matrices is allowed by <a href='eco.convert.html'>eco.convert</a> and <a href='eco.format.html'>eco.format</a>. Tools for computation of NDVI in Landsat imaginery, 
post-process of rasters and temporal analysis can be found in <a href='eco.NDVI.html'>eco.NDVI</a>, <a href='eco.NDVI.post.html'>eco.NDVI.post</a> and <a href='eco.theilsen.html'>eco.theilsen</a>. Other useful functions are <a href='aue.sort.html'>aue.sort</a> (for ordering alleles), <a href='eco.alfreq.html'>eco.alfreq</a> (to comput histograms of allelic frequencies for detection of bottlenecks), <a href='eco.post.geneland.html'>eco.post.geneland</a> and <a href='eco.pairtest.html'>eco.pairtest</a>.</p>
<p>The results obtained with the main functions defined in EcoGenetics are
object of class S4. As a default characteristic of the package design,
these objects have a "show"" method for a general overview of the results, and methods to extract the results stored in slots (generic <a href='EcoGenetics-accessors.html'>accessors</a> and double square brackets ("[[") definitions).</p>
<p>For storing and pre-processing the data to analyze, the package 
defines a special class: the class <code><a href='ecogen.html'>ecogen</a></code>.</p> 
<p><b><em>II. STRUCTURE OF ECOGEN OBJECTS: HANDLING AND INTEGRATING INFORMATION</em></b></p>
    <p>Ecological genetics research requires the integration of data originated in different sources. The class <code><a href='ecogen.html'>ecogen</a></code> has been designed for handling multidimensional data. Its basic structure is the following:</p>
<p>- An <strong>XY</strong> slot, storing a data frame with geographic coordinates.</p>
<p>- A <strong>P</strong> slot, storing a phenotypic data frame.</p>
<p>- A <strong>G</strong> slot, storing a genotypic data frame.</p>
<p>- An <strong>A</strong> slot containing as allelic frequencies the information of G (only
available for codominant markers.)</p>
<p>- An <strong>E</strong> slot, storing an environmental data frame.</p>
<p>- A <strong>S</strong> slot, storing a data frame with classes assigned to the 
individuals.</p>
    <p>- A <strong>C</strong> slot, for a custom data frame.</p>
<p>- An <strong>OUT</strong> slot, containing a list for the storage of the results.</p>
    <p>For dominant (presence-absence) markers, the slot A is empty.</p>
<p><b><em>III. A BRIEF OVERVIEW OF THE DATA HANDLING METHODS DEFINED FOR ECOGEN OBJECTS</em></b></p>
    <p>The construction of a new "ecogen" object from a data frame is made with the homonymous function.
    <em>library("EcoGenetics")</em>
    <em>data(eco.test)</em>
    <em>eco &lt;- ecogen(XY = coordinates, P = phenotype, G = genotype, E = environment, S = structure, order.G = TRUE)</em></p>
    <p># The following methods can be used with ecogen objects:</p>
<p># - - - - - -
    <b># 1. subsetting/ordering by row-method, using single square brackets ("[")</b>
    <em>eco.sub &lt;- eco[1:50]</em>
    <em>eco.ord &lt;- eco[c(50:1)]</em>
    <em>eco.ord[["XY"]] # object ordered and subset rows.
This is a powerful method for handling the stack of ecogen data frames</em>
    <em>eco &lt;- eco[order(eco[["S"]][, 1])] #ordering S[, 1] and then the object</em>
    <b># 2. <a href='eco.merge.html'>merging-method</a>, for two objects</b>
    <em>eco1 &lt;- eco</em>
    <em>merged &lt;- eco.merge(eco, eco1)</em>
    <b># 3. <a href='eco.subset.html'>subsetting-method</a>, in reference to a group in the S slot (in this case, "1")</b>
    <em>eco.subS &lt;- eco.subset(eco,"pop", 1)</em></p>
<p><b># 4. <a href='eco.rbind.html'>binding by row-method</a> (duplicated row names not allowed)</b>
    <em>eco2 &lt;- eco; rownames(eco2[["P"]]) &lt;-226:450</em>
    <em>eco.r &lt;- eco.rbind(eco, eco2)</em>
    <b># 5. <a href='eco.cbind.html'>binding by column-method</a></b>
    <em>eco.c &lt;- eco.cbind(eco, eco1)</em></p>
<p><b># 6. get-method using generic <a href='EcoGenetics-accessors.html'>accessors</a> and "[[" (equivalent methods):</b>
    <em>ecoslot.XY(eco) ; eco[["XY"]]</em>
    <em>ecoslot.P(eco); eco[["P"]]</em>
    <em>ecoslot.G(eco); eco[["G"]]</em>
    <em>ecoslot.A(eco); eco[["A"]]  (only available for codominant data</em>
    <em>ecoslot.E(eco); eco[["E"]]</em>
    <em>ecoslot.S(eco); eco[["S"]]</em>
    <em>ecoslot.C(eco); eco[["C"]]</em>
    <em>ecoslot.OUT(eco); eco[["OUT"]]</em></p>
<p><b># 7. set-method using <a href='EcoGenetics-accessors.html'>accessors</a> and "[[" (equivalent methods):</b>
    <em>eco.temp &lt;- ecogen(XY = coordinates, P = phenotype)</em>
    <em>eco.temp</em>
    <em>ecoslot.G(eco.temp, order.G = TRUE) &lt;- genotype</em></p>
<p>## this is equivalent, in square brackets notation, to:
    <em>eco[["G", order.G=TRUE]] &lt;- genotype</em>
    <em>ecoslot.E(eco.temp) &lt;- environment</em></p>
<p>## identical to eco[["E"]] &lt;- environment
    <em>ecoslot.S(eco.temp) &lt;- structure</em></p>
<p>## identical to eco[["S"]] &lt;- structure
    <em>eco.temp</em>
    <b># 8. appending-method (storing information generated via <a href='EcoGenetics-accessors.html'>accessors)</a></b></p>
<p>## fitting a multiple linear regression model:
    <em>linear.analysis &lt;- eco.lmtree(eco[["P"]], eco[["E"]], "mlm")</em></p>
<p>## storing the results:
    <em>ecoslot.OUT(eco) &lt;- linear.analysis</em>
    <em>eco</em></p>
<p>## Storing multiple result at once:
    <em>a &lt;- c(1:10)</em>
    <em>b &lt;- c(2:30)</em>
    <em>ecoslot.OUT(eco) &lt;- list(eco, a, b)</em>
    <em>eco</em></p>
<p>## the use of the accessor OUT has its equivalence in double
square brackets notation:
    <em>eco[["OUT"]] &lt;- list(eco, a, b)</em></p>
    <p>## summary table
    <em>ecoslot.OUT(eco)</em></p>  
<p>## the data frame shows the results stored in alphabetical order and their classes. The specification of a second name, return the corresponding stored object:
    <em>ecoslot.OUT(eco, "a")</em></p>
<p>## note that the append method is a particular case of the
## set method with accessors / "[[" for the slot OUT.
    <b># 9. <a href='eco.remove.html'>removing-method</a></b></p>
<p>## removing objects a and b from eco
    <em>eco &lt;- eco.remove(eco, a, b)</em></p>
    <p>- - - - - -</p>
<p>The workspace can be <a href='eco.clear.html'>cleared</a>, only storing the desired object:
    <em>ls()</em>
    <em>eco.clear(eco)</em>
    <em>ls()</em></p>
<p>The cleaner and compact option, is to save the object as RDS,
which is a way to save R objects as a file:
    <em>saveRDS(eco, "myObject.rds")</em></p>
<p>The object can be loaded in the future with the function readRDS:
    <em>eco &lt;- readRDS("myObject.rds")</em></p>
<p>----------------------- o -----------------------</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
    <p>Anselin L. 1995. Local indicators of spatial association-LISA. Geographical analysis. 27: 93-115.</p>
<p>Borcard D., F. Gillet, and P. Legendre. 2011. Numerical ecology with R. Springer Science &amp; Business Media.</p>
<p>Chander G., B. Markham, and D. Helder. 2009. Summary of current radiometric calibration coefficients for Landsat MSS, TM, ETM+, and EO-1 ALI sensors. Remote sensing of environment, 113: 893-903.</p>
<p>Chavez P. 1989. Radiometric calibration of Landsat Thematic Mapper multispectral images. Photogrammetric Engineering and Remote Sensing, 55: 1285-1294.</p>
<p>Chavez P. 1996. Image-based atmospheric corrections-revisited and improved. Photogrammetric engineering and remote sensing, 62: 1025-1035.</p>
<p>Double M., R. Peakall, N. Beck, and Y. Cockburn. 2005. Dispersal, philopatry, and infidelity: dissecting local genetic structure in superb fairy-wrens (Malurs cyaneus). Evolution 59: 625-635.</p>
<p>Dray S., and A., Dufour. 2007. The ade4 package: implementing the duality diagram for ecologists. Journal of statistical software, 22: 1-20.</p> 
<p>Freedman D., and P. Diaconis. 1981. On the histogram as a density estimator: L 2 theory. Probability theory and related fields, 57: 453-476.</p>  
<p>Geary R. 1954. The contiguity ratio and statistical mapping. The incorporated statistician, 115-146.</p>
<p>Getis A., and J. Ord. 1992. The analysis of spatial association by use of distance statistics. Geographical analysis, 24: 189-206.</p>
<p>Goslee S. 2011. Analyzing remote sensing data in R: the landsat package. Journal of Statistical Software, 43: 1-25.</p>
<p>Goudet J. 2005. Hierfstat, a package for R to compute and test hierarchical F-statistics. Molecular Ecology Notes, 5: 184-186.</p> 
<p>Guillot G., F. Mortier and A. Estoup. 2005. GENELAND: a computer package for landscape genetics. Molecular Ecology Notes, 5: 712-715.</p> 
<p>Jombart T. 2008. adegenet: a R package for the multivariate analysis of genetic markers. Bioinformatics, 24: 1403-1405.</p>
<p>Kalisz S., J. Nason, F.M. Handazawa, and S. Tonsor. 2001. Spatial population genetic structure in Trillium grandiflorum:  the roles of dispersal, mating, history, and selection.  Evolution 55: 1560-1568.</p>
<p>Legendre P., and L. Legendre. 2012. Numerical ecology. Third English edition .Elsevier Science, Amsterdam, Netherlands.</p>
<p>Lichstein J., T. Simons, S. Shriner, and K. Franzreb. 2002. Spatial autocorrelation and autoregressive models in ecology. Ecological monographs, 72: 445-463.</p>
<p>Loiselle B., V. Sork, J. Nason, and C. Graham. 1995. Spatial genetic structure of a tropical understory shrub, Psychotria officinalis (Rubiaceae). American Journal of Botany 1420-1425.</p>
<p>Moran P. 1950. Notes on continuous stochastic phenomena. Biometrika, 17-23.</p>
<p>Oden, N., 1984. Assessing the significance of a spatial correlogram. 
Geographical Analysis, 16: 1-16.</p>
<p>Oden N., and R. Sokal. 1986. Directional autocorrelation: an extension of spatial correlograms to two dimensions. Systematic Zoology, 35: 608-617.</p>
<p>Ord J., and A. Getis. 1995. Local spatial autocorrelation statistics: distributional issues and an application. Geographical analysis, 27: 286-306.</p>
<p>Reich R., R. Czaplewski and W. Bechtold. 1994. Spatial cross-correlation of undisturbed, natural shortleaf pine stands in northern Georgia. Environmental and Ecological Statistics, 1: 201-217.</p>
<p>Sokal R., and N. Oden 1978. Spatial autocorrelation in biology: 1. Methodology. Biological journal of the Linnean Society, 10: 199-228.</p>
<p>Sokal R., and N. Oden. 1978. Spatial autocorrelation in biology. 2. Some biological implications and four applications of evolutionary and ecological interest. Biological Journal of the Linnean Society, 10: 229-49.</p>
<p>Sokal R. 1979. Ecological parameters inferred from spatial correlograms. In: G. Patil and M. Rosenzweig, editors. Contemporary Quantitative Ecology and elated Ecometrics. International Co-operative Publishing House: Fairland, MD, pp. 167-96.</p>
<p>Sokal R. 1986. Spatial data analysis and historical processes. In: E. Diday, Y. Escoufier, L. Lebart, J. Pages, Y. Schektman, and R. Tomassone, editors. Data analysis and informatics, IV. North-Holland, Amsterdam, The Netherlands, pp. 29-43.</p>
<p>Sokal R., N. Oden and B. Thomson. 1998. Local spatial autocorrelation in a biological model. Geographical Analysis, 30: 331-354.</p>
<p>Sokal R., and B. Thomson. 2006. Population structure inferred by local spatial autocorrelation: an example from an Amerindian tribal population. American journal of physical anthropology, 129: 121-131.</p>
<p>Song C., C. Woodcock, K. Seto, M. Lenney and S. Macomber. 2001. Classification and change detection using Landsat TM data: when and how to correct atmospheric effects?. Remote sensing of Environment, 75: 230-244.</p>
<p>Sturges H. 1926. The choice of a class interval. Journal of the American Statistical Association, 21: 65-66.</p>
<p>Tucker C. 1979. Red and photographic infrared linear combinations for monitoring vegetation. Remote sensing of Environment, 8: 127-150.</p>
<p>Vekemans, X., and O. Hardy. 2004. New insights from fine-scale spatial genetic structure analyses in plant populations. Molecular Ecology, 
13: 921-935.</p>
<p>Wu. 1986. Jackknife, bootstrap and other resampling methods in regression analysis. the Annals of Statistics, 1261-1295.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
 </div><span class='co'># NOT RUN {</span>


<span class='co'>#---Detrending spatial data with polynomial interpolation---#</span>


<span class='fu'>data</span>(<span class='no'>eco2</span>)

<span class='co'>## original data</span>
<span class='no'>data1</span> <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='no'>eco2</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>1</span>], <span class='fl'>30</span>, <span class='fl'>30</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='no'>data1</span>)

<span class='co'>## original data + trend</span>
<span class='no'>data2</span> <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='no'>eco2</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>2</span>], <span class='fl'>30</span>, <span class='fl'>30</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='no'>data2</span>)

<span class='co'>## data detrending</span>
<span class='no'>data2.det</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.detrend.html'>eco.detrend</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco2</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>2</span>], <span class='kw'>XY</span> <span class='kw'>=</span>  <span class='no'>eco2</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>degree</span> <span class='kw'>=</span>  <span class='fl'>1</span>)
<span class='no'>data2.det</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.RES</a></span>(<span class='no'>data2.det</span>)
<span class='no'>data2.det</span> <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='no'>data2.det</span>$<span class='no'>df1</span>, <span class='fl'>30</span>, <span class='fl'>30</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='no'>data2.det</span>)


<span class='co'>#---Multiple Linear Regression fit---#</span>


<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>mymod</span> <span class='kw'>&lt;-</span> <span class='st'>"E1+E2*E3"</span>
<span class='no'>mod</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lmtree.html'>eco.lmtree</a></span>(<span class='kw'>df1</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>df2</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]],
<span class='kw'>analysis</span> <span class='kw'>=</span> <span class='st'>"mlm"</span>, <span class='kw'>mod.class</span> <span class='kw'>=</span> <span class='no'>mymod</span>)
<span class='fu'>summary</span>(<span class='no'>mod</span>)


<span class='co'>#---Multiple Conditional Inference Trees---#</span>


<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>mymod</span> <span class='kw'>&lt;-</span> <span class='st'>"E1+E2*E3"</span>
<span class='no'>mod</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lmtree.html'>eco.lmtree</a></span>(<span class='kw'>df1</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]], <span class='kw'>df2</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]],
<span class='kw'>analysis</span> <span class='kw'>=</span> <span class='st'>"mctree"</span>, <span class='kw'>mod.class</span> <span class='kw'>=</span> <span class='no'>mymod</span>, <span class='kw'>fact</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"S"</span>]]$<span class='no'>structure</span>)
<span class='fu'>summary</span>(<span class='no'>mod</span>)


<span class='co'>#---Global spatial analysis---#</span>


<span class='co'>## Moran's I </span>

<span class='co'>### one test</span>
<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>con</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"circle"</span>, <span class='kw'>d1</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>d2</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>global</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con</span>, , <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>200</span>)
<span class='no'>global</span>

<span class='fu'>require</span>(<span class='no'>adegenet</span>)
<span class='no'>con2</span><span class='kw'>&lt;-</span><span class='fu'>chooseCN</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>type</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>result.type</span> <span class='kw'>=</span> <span class='st'>"listw"</span>, <span class='kw'>plot.nb</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>global</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con2</span>, , <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>200</span>)
<span class='no'>global</span>

<span class='co'>### multiple tests</span>
<span class='no'>con</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"circle"</span>, <span class='kw'>d1</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>d2</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>global</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]], <span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con</span>, , <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>200</span>)
<span class='no'>global</span>


<span class='co'>## Geary's C </span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>global.C</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"C"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>200</span>)
<span class='no'>global.C</span>


<span class='co'>## Bivariate's Moran's Ixy</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>global.Ixy</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]][, <span class='fl'>1</span>],
<span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"CC"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>200</span>)
<span class='no'>global.Ixy</span>


<span class='co'>## Join-Count</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>global.JC</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.gsa.html'>eco.gsa</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='fl'>2</span>* <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"A"</span>]][, <span class='fl'>1</span>], <span class='kw'>ncod</span> <span class='kw'>=</span> <span class='fl'>1</span>,
<span class='kw'>con</span> <span class='kw'>=</span> <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"JC"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>5</span>)
<span class='no'>global.JC</span>


<span class='co'># Mantel test</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'><a href='eco.mantel.html'>eco.mantel</a></span>(<span class='kw'>d1</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]]), <span class='kw'>d2</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]]), <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)


<span class='co'>## Partial Mantel test</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'><a href='eco.mantel.html'>eco.mantel</a></span>(<span class='kw'>d1</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]]), <span class='kw'>d2</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]]),
<span class='kw'>dc</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]]), <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)


<span class='co'>#---Local spatial analysis---#</span>


<span class='co'>## Getis-Ord's G*</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>con</span><span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"knearest"</span>,  <span class='kw'>k</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>self</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='co'>### self = TRUE for G*</span>
<span class='no'>getis.ak</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"G*"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>,
<span class='kw'>adjust</span> <span class='kw'>=</span> <span class='st'>"none"</span>)
<span class='no'>getis.ak</span>

<span class='co'>### to plot the results, the function "eco.lsa" calls "eco.rankplot"</span>
<span class='co'>### (see ?eco.rankplot) when test = "permutation" and "eco.forestplot"</span>
<span class='co'>### (see ?eco.forestplot) when test = "bootstrap"</span>

<span class='no'>p</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>getis.ak</span>)   <span class='co'>###  rankplot graph</span>
<span class='no'>p</span>                     <span class='co'>###  points with colors of the color-scale:</span>
                      <span class='co'>###  points with P &lt; 0.05. Yellow points : </span>
                      <span class='co'>###  points with P &gt; 0.05</span>
<span class='no'>p</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>getis.ak</span>, <span class='kw'>significant</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)
<span class='no'>p</span>                     <span class='co'>###  all points have a color of the color-scale</span>

<span class='co'>### bootstrap example</span>
<span class='no'>getis.akb</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"G*"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>,
<span class='kw'>test</span> <span class='kw'>=</span> <span class='st'>"bootstrap"</span>)
<span class='no'>p</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>getis.akb</span>)     <span class='co'>### forestplot graph</span>
<span class='no'>p</span> + <span class='kw pkg'>ggplot2</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme'>theme_bw</a></span>()  <span class='co'>### the plot can be modified with ggplot2</span>
                         <span class='co'>### In this case, the background color is modified </span>


<span class='co'>## Getis-Ord's G</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>con</span><span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"knearest"</span>, <span class='kw'>k</span> <span class='kw'>=</span> <span class='fl'>4</span>)
<span class='co'>### self = FALSE for G</span>
<span class='no'>getis</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"G"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>, <span class='kw'>adjust</span> <span class='kw'>=</span> <span class='st'>"none"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>getis</span>)


<span class='co'>## Local Moran's I</span>

 <span class='co'>#-------------------------</span>
 <span class='co'># TESTING PHENOTYPIC DATA-</span>
 <span class='co'>#-------------------------</span>

 <span class='no'>con</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"knearest"</span>,  <span class='kw'>k</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>row.sd</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
 <span class='co'># row standardized weights = TRUE</span>

 <span class='co'># test for the first trait of the data frame P </span>
 <span class='no'>localmoran</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)

 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>localmoran</span>)

 <span class='co'># test for several variables</span>

 <span class='no'>all.traits</span> <span class='kw'>&lt;-</span> <span class='fu'>apply</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]], <span class='fl'>2</span>, <span class='no'>eco.lsa</span>,  <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)

 <span class='co'># Observed statistic and P-values tables (individuals x traits)</span>
 <span class='no'>stat.P</span> <span class='kw'>&lt;-</span> <span class='fu'>sapply</span>(<span class='no'>all.traits</span>, <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fu'>return</span>(<span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.OUT</a></span>(<span class='no'>x</span>)[,<span class='fl'>1</span>]))
 <span class='no'>pval.P</span> <span class='kw'>&lt;-</span> <span class='fu'>sapply</span>(<span class='no'>all.traits</span>, <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fu'>return</span>(<span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.OUT</a></span>(<span class='no'>x</span>)[,<span class='fl'>4</span>]))

 <span class='co'># Plot of the phenotypic spatial patterns</span>

 <span class='fu'>par</span>(<span class='kw'>mfrow</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>2</span>,<span class='fl'>4</span>))
 <span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>8</span>) {
 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='fu'>matrix</span>(<span class='no'>stat.P</span>[,<span class='no'>i</span>], <span class='fl'>15</span>,<span class='fl'>15</span>))
 }

 <span class='fu'>par</span>(<span class='kw'>mfrow</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>2</span>,<span class='fl'>4</span>))
 <span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>8</span>) {
 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='fu'>matrix</span>(<span class='no'>pval.P</span>[,<span class='no'>i</span>], <span class='fl'>15</span>,<span class='fl'>15</span>))
 }


 <span class='co'>#-------------------------</span>
 <span class='co'># TESTING GENOTYPIC DATA-</span>
 <span class='co'>#-------------------------</span>

 <span class='co'># eco[["A"]] is a matrix with the genetic data of "eco"</span>
 <span class='co'># as frequencies for each allele in each individual</span>
 <span class='co'># (use [["G"]] for presence-absence data).</span>

 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/headtail'>head</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"A"</span>]])      <span class='co'># head of the matrix - 40 alleles</span>

 <span class='no'>con</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"knearest"</span>,  <span class='kw'>k</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>row.sd</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
 <span class='co'># row standardized weights = TRUE</span>

 <span class='co'># test for a single allele</span>
 <span class='no'>localmoran.geno</span> <span class='kw'>&lt;-</span>  <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"A"</span>]][, <span class='fl'>32</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)

 <span class='co'># test for several alleles -  40 alleles (it runs in less than 1 min </span>
 <span class='co'># for 99 simulations per allele;  999 simulations takes ~ 11 s per allele, </span>
 <span class='co'># less than 8 min in total.) </span>
 <span class='no'>all.alleles</span> <span class='kw'>&lt;-</span> <span class='fu'>apply</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"A"</span>]], <span class='fl'>2</span>, <span class='no'>eco.lsa</span>,  <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)

 <span class='co'># plot all alleles to get an overview of the spatial patterns</span>
 <span class='fu'>lapply</span>(<span class='no'>all.alleles</span>, <span class='no'>plot</span>)

 <span class='co'># Observed statistic and P-values tables (individuals x loci)</span>
 <span class='no'>stat.G</span> <span class='kw'>&lt;-</span> <span class='fu'>sapply</span>(<span class='no'>all.alleles</span>, <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fu'>return</span>(<span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.OUT</a></span>(<span class='no'>x</span>)[,<span class='fl'>1</span>]))
 <span class='no'>pval.G</span> <span class='kw'>&lt;-</span> <span class='fu'>sapply</span>(<span class='no'>all.alleles</span>, <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fu'>return</span>(<span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.OUT</a></span>(<span class='no'>x</span>)[,<span class='fl'>4</span>]))

 <span class='co'># counting individuals with P &lt; 0.05 for each allele </span>
 <span class='co'># (5 * 225 /100 ~  12   significant tests by random)</span>
 <span class='no'>signif</span> <span class='kw'>&lt;-</span> <span class='fu'>lapply</span>(<span class='no'>all.alleles</span>, <span class='kw'>function</span>(<span class='no'>x</span>) <span class='fu'>sum</span>(<span class='fu'><a href='EcoGenetics-accessors.html'>ecoslot.OUT</a></span>(<span class='no'>x</span>)[,<span class='fl'>4</span>] <span class='kw'>&lt;</span> <span class='fl'>0.05</span>))
 <span class='no'>signif</span> <span class='kw'>&lt;-</span> <span class='fu'>unlist</span>(<span class='no'>signif</span>)

 <span class='co'># filtering alleles, loci with &gt; 12 significant individual tests</span>

 <span class='no'>A.local</span> <span class='kw'>&lt;-</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"A"</span>]][, <span class='no'>signif</span> <span class='kw'>&gt;</span> <span class='fl'>12</span>]     <span class='co'>#filtered matrix</span>
 <span class='no'>stat.G.f</span> <span class='kw'>&lt;-</span> <span class='no'>stat.G</span>[, <span class='no'>signif</span> <span class='kw'>&gt;</span> <span class='fl'>12</span>]
 <span class='no'>pval.G.f</span> <span class='kw'>&lt;-</span> <span class='no'>pval.G</span>[, <span class='no'>signif</span> <span class='kw'>&gt;</span> <span class='fl'>12</span>]

 <span class='co'># Plot of the genotypic spatial patterns</span>

 <span class='co'># one plot possibility, using the EcoGenetics method &lt;rankplot&gt;</span>
 <span class='no'>all.local</span> <span class='kw'>&lt;-</span> <span class='no'>all.alleles</span>[<span class='no'>signif</span> <span class='kw'>&gt;</span> <span class='fl'>12</span>]
 <span class='fu'>lapply</span>(<span class='no'>all.local</span>, <span class='no'>plot</span>)

 <span class='co'># other plot possibility, using &lt;image&gt;</span>
 <span class='fu'>par</span>(<span class='kw'>mfrow</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>3</span>,<span class='fl'>4</span>))
 <span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>12</span>) {
 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='fu'>matrix</span>(<span class='no'>stat.G</span>[,<span class='no'>i</span>], <span class='fl'>15</span>,<span class='fl'>15</span>))
 }

 <span class='fu'>par</span>(<span class='kw'>mfrow</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>3</span>,<span class='fl'>4</span>))
 <span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>12</span>) {
 <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/image'>image</a></span>(<span class='fu'>matrix</span>(<span class='no'>pval.G</span>[,<span class='no'>i</span>], <span class='fl'>15</span>,<span class='fl'>15</span>))
 }


<span class='co'>## Local Geary's C</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>con</span><span class='kw'>&lt;-</span> <span class='fu'><a href='eco.weight.html'>eco.weight</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"knearest"</span>,  <span class='kw'>k</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>row.sd</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='co'>### row standardized weights = TRUE</span>
<span class='no'>localgeary</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.lsa.html'>eco.lsa</a></span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>1</span>], <span class='no'>con</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"C"</span>, <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>,
<span class='kw'>adjust</span> <span class='kw'>=</span> <span class='st'>"none"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>localgeary</span>)


<span class='co'>#---Moran's I, Geary's C and bivariate Moran's Ixy correlograms---#</span>


<span class='co'>## Moran's I correlogram</span>

<span class='co'>### single test</span>
<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>moran</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.correlog.html'>eco.correlog</a></span>(<span class='kw'>Z</span><span class='kw'>=</span><span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>1</span>], <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>,
<span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>moran</span>)

<span class='co'>### multiple tests</span>
<span class='no'>moran2</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.correlog.html'>eco.correlog</a></span>(<span class='kw'>Z</span><span class='kw'>=</span><span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]], <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"I"</span>,
<span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>moran2</span>, <span class='kw'>var</span> <span class='kw'>=</span><span class='st'>"P2"</span>) <span class='co'>## single plots</span>
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>moran2</span>, <span class='kw'>var</span> <span class='kw'>=</span><span class='st'>"P3"</span>) <span class='co'>## single plots</span>

<span class='no'>graf</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>moran2</span>, <span class='kw'>meanplot</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)  <span class='co'>## multiple plot with mean correlogram</span>
                                       <span class='co'>## and jackknifed confidence intervals.</span>

<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>graf</span><span class='kw'>[[</span><span class='fl'>1</span>]])
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>graf</span><span class='kw'>[[</span><span class='fl'>2</span>]])

<span class='co'>### correlogram plots support the use of ggplot2 syntax</span>
<span class='no'>moranplot</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>moran2</span>, <span class='kw'>var</span> <span class='kw'>=</span><span class='st'>"P3"</span>) + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme'>theme_bw</a></span>() + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/theme'>theme</a></span>(<span class='kw'>legend.position</span><span class='kw'>=</span><span class='st'>"none"</span>)
<span class='no'>moranplot</span>


<span class='co'>## Geary's C correlogram</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>geary</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.correlog.html'>eco.correlog</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>1</span>], <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"C"</span>,
<span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>geary</span>)


<span class='co'>## Moran's Ixy cross-correlogram</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>cross</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.correlog.html'>eco.correlog</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][,<span class='fl'>1</span>], <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]][, <span class='fl'>1</span>],
<span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"CC"</span>, <span class='kw'>int</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>smax</span> <span class='kw'>=</span> <span class='fl'>15</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>cross</span>)


<span class='co'>#---Mantel and partial Mantel correlograms---#</span>


<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)

<span class='co'>## Mantel correlogram</span>

<span class='no'>corm</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.cormantel.html'>eco.cormantel</a></span>(<span class='kw'>M</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]]), <span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>,<span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>7</span>, <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]],
<span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>corm</span>)

<span class='no'>corm</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.cormantel.html'>eco.cormantel</a></span>(<span class='kw'>M</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]]), <span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>,<span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>7</span>, <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]],
<span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>, <span class='kw'>test</span> <span class='kw'>=</span> <span class='st'>"bootstrap"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>corm</span>)



<span class='co'>## Partial Mantel correlogram</span>

<span class='no'>corm</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.cormantel.html'>eco.cormantel</a></span>(<span class='kw'>M</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]]), <span class='kw'>MC</span> <span class='kw'>=</span> <span class='fu'>dist</span>(<span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"E"</span>]]),
<span class='kw'>size</span><span class='kw'>=</span><span class='fl'>1000</span>, <span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>7</span>, <span class='kw'>XY</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]], <span class='kw'>nsim</span> <span class='kw'>=</span> <span class='fl'>99</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>corm</span>)

<span class='co'>### correlogram plots support the use of ggplot2 syntax</span>
<span class='no'>mantelplot</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>corm</span>) + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme'>theme_bw</a></span>() + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/theme'>theme</a></span>(<span class='kw'>legend.position</span><span class='kw'>=</span><span class='st'>"none"</span>)
<span class='no'>mantelplot</span>


<span class='co'>#---Empirical variogram---#</span>


<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>require</span>(<span class='no'>ggplot2</span>)
<span class='no'>variog</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.variogram.html'>eco.variogram</a></span>(<span class='kw'>Z</span> <span class='kw'>=</span> <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"P"</span>]][, <span class='fl'>2</span>],<span class='kw'>XY</span> <span class='kw'>=</span>  <span class='no'>eco</span><span class='kw'>[[</span><span class='st'>"XY"</span>]])
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>variog</span>)

<span class='co'>### variogram plots support the use of ggplot2 syntax</span>
<span class='no'>variogplot</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>variog</span>) + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme'>theme_bw</a></span>() + <span class='fu'><a href='http://www.rdocumentation.org/packages/ggplot2/topics/theme'>theme</a></span>(<span class='kw'>legend.position</span><span class='kw'>=</span><span class='st'>"none"</span>)
<span class='no'>variogplot</span>


<span class='co'>#---Computing NDVI with atmospheric correction</span>
<span class='co'>#---over a time series, and estimation of the temporal mean---#</span>

<span class='fu'>require</span>(<span class='no'>raster</span>)
<span class='fu'>data</span>(<span class='no'>tab</span>)
<span class='fu'>data</span>(<span class='no'>eco3</span>)
<span class='fu'>set.seed</span>(<span class='fl'>6</span>)

<span class='no'>temp</span> <span class='kw'>&lt;-</span> <span class='fu'>list</span>()

<span class='co'>## we create 4 simulated rasters for the data included in the object tab:</span>

<span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>4</span>) {
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='fu'>runif</span>(<span class='fl'>19800</span>, <span class='fl'>0</span>, <span class='fl'>254</span>)
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]], <span class='fl'>180</span>, <span class='fl'>110</span>)
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/raster'>raster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]], <span class='kw'>crs</span><span class='kw'>=</span><span class='st'>"+proj=utm"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/extent'>extent</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]])<span class='kw'>&lt;-</span><span class='fu'>c</span>(<span class='fl'>3770000</span>, <span class='fl'>3950000</span>, <span class='fl'>6810000</span>, <span class='fl'>6920000</span>)
}

<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/writeRaster'>writeRaster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='fl'>1</span>]], <span class='st'>"20040719b4.tif"</span>, <span class='kw'>overwrite</span> <span class='kw'>=</span> <span class='no'>T</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/writeRaster'>writeRaster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='fl'>2</span>]], <span class='st'>"20040719b3.tif"</span>, <span class='kw'>overwrite</span> <span class='kw'>=</span> <span class='no'>T</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/writeRaster'>writeRaster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='fl'>3</span>]], <span class='st'>"20091106b4.tif"</span>, <span class='kw'>overwrite</span> <span class='kw'>=</span> <span class='no'>T</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/writeRaster'>writeRaster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='fl'>4</span>]], <span class='st'>"20091106b3.tif"</span>, <span class='kw'>overwrite</span> <span class='kw'>=</span> <span class='no'>T</span>)

<span class='co'>## Computing NDVI images:</span>

<span class='fu'><a href='eco.NDVI.html'>eco.NDVI</a></span>(<span class='no'>tab</span>, <span class='st'>"COST"</span>, <span class='st'>"NDVI"</span>, <span class='st'>"LT5"</span>)

<span class='co'>## Mean NDVI image computed over the NDVI images that we calculated:</span>

<span class='fu'><a href='eco.NDVI.post.html'>eco.NDVI.post</a></span>(<span class='no'>tab</span>, <span class='st'>"COST"</span>, <span class='st'>"NDVI"</span>, <span class='kw'>what</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"mean"</span>, <span class='st'>"var"</span>))
<span class='no'>mean.ndvi</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/raster'>raster</a></span>(<span class='st'>"NDVI.COST.mean.tif"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>mean.ndvi</span>)

<span class='co'>## Extraction of the mean NDVI for each point in the object eco and plot of the data:</span>

<span class='no'>ndvi</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/extract'>extract</a></span>(<span class='no'>mean.ndvi</span>, <span class='no'>eco3</span><span class='kw'>[[</span><span class='st'>"XY"</span>]])
<span class='no'>ndvi</span><span class='kw'>&lt;-</span> <span class='fu'><a href='aue.rescale.html'>aue.rescale</a></span>(<span class='no'>ndvi</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>eco3</span><span class='kw'>[[</span><span class='st'>"XY"</span>]][, <span class='fl'>1</span>], <span class='no'>eco3</span><span class='kw'>[[</span><span class='st'>"XY"</span>]][, <span class='fl'>2</span>], <span class='kw'>col</span><span class='kw'>=</span><span class='fu'>rgb</span>(<span class='no'>ndvi</span>, <span class='fl'>0</span>, <span class='fl'>0</span>),
<span class='kw'>pch</span><span class='kw'>=</span><span class='fl'>15</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>"Mean NDVI"</span>, <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"X"</span>, <span class='kw'>ylab</span>  <span class='kw'>=</span> <span class='st'>"Y"</span>)


<span class='co'>#---Theil-sen estimation for a raster---#</span>

<span class='fu'>require</span>(<span class='st'>"raster"</span>)
<span class='fu'>set.seed</span>(<span class='fl'>6</span>)

<span class='no'>temp</span> <span class='kw'>&lt;-</span> <span class='fu'>list</span>()

<span class='kw'>for</span>(<span class='no'>i</span> <span class='kw'>in</span> <span class='fl'>1</span>:<span class='fl'>100</span>) {
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='fu'>runif</span>(<span class='fl'>36</span>,-<span class='fl'>1</span>, <span class='fl'>1</span>)
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]], <span class='fl'>6</span>, <span class='fl'>6</span>)
<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]] <span class='kw'>&lt;-</span> <span class='kw pkg'>raster</span><span class='kw ns'>::</span><span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/raster'>raster</a></span>(<span class='no'>temp</span><span class='kw'>[[</span><span class='no'>i</span>]])
}

<span class='no'>temp</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/brick'>brick</a></span>(<span class='no'>temp</span>)

<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/writeRaster'>writeRaster</a></span>(<span class='no'>temp</span>,<span class='st'>"temporal.tif"</span>, <span class='kw'>overwrite</span><span class='kw'>=</span><span class='no'>T</span>)
<span class='fu'>rm</span>(<span class='no'>temp</span>)
<span class='no'>ndvisim</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/brick'>brick</a></span>(<span class='st'>"temporal.tif"</span>)
<span class='no'>date</span> <span class='kw'>&lt;-</span> <span class='fu'>seq</span>(<span class='kw'>from</span> <span class='kw'>=</span> <span class='fl'>1990.1</span>, <span class='kw'>length.out</span> <span class='kw'>=</span> <span class='fl'>100</span>, <span class='kw'>by</span> <span class='kw'>=</span> <span class='fl'>0.2</span>)
<span class='fu'><a href='eco.theilsen.html'>eco.theilsen</a></span>(<span class='no'>ndvisim</span>, <span class='no'>date</span>)
<span class='no'>pvalue</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/raster'>raster</a></span>(<span class='st'>"pvalue.tif"</span>)
<span class='no'>slope</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/raster'>raster</a></span>(<span class='st'>"slope.tif"</span>)
<span class='fu'>par</span>(<span class='kw'>mfrow</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='fl'>1</span>, <span class='fl'>2</span>))
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>pvalue</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>"p-value"</span>)
<span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/plot'>plot</a></span>(<span class='no'>slope</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>"slope"</span>)


<span class='co'>#---Conversion of ecogen data to several formats and format tool---#</span>


<span class='co'>## Interconverting an ecogen genetic data frame among several formats </span>

<span class='fu'>data</span>(<span class='no'>eco3</span>)

<span class='co'>### One allele per column</span>
<span class='no'>loc2al</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.convert.html'>eco.convert</a></span>(<span class='no'>eco3</span><span class='kw'>[[</span><span class='st'>"G"</span>]], <span class='st'>"matrix"</span>, <span class='st'>"alleles.matrix"</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>loc2al</span>

<span class='co'>### Inverse operation (collapse alleles into locus)</span>
<span class='no'>al2loc</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.convert.html'>eco.convert</a></span>(<span class='no'>loc2al</span>, <span class='st'>"alleles.matrix"</span>, <span class='st'>"matrix"</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>)
<span class='no'>al2loc</span>

<span class='co'>### Separating alleles with a character string</span>
<span class='no'>loc2loc</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.convert.html'>eco.convert</a></span>(<span class='no'>eco3</span><span class='kw'>[[</span><span class='st'>"G"</span>]], <span class='st'>"matrix"</span>, <span class='st'>"matrix"</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>sep.out</span> <span class='kw'>=</span> <span class='st'>"/"</span>)
<span class='no'>loc2loc</span>

<span class='co'>### Inverse operation (removing separator)</span>
<span class='no'>loc2loc.nosep</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.convert.html'>eco.convert</a></span>(<span class='no'>loc2loc</span>, <span class='st'>"matrix"</span>, <span class='st'>"matrix"</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>sep.in</span> <span class='kw'>=</span> <span class='st'>"/"</span>, <span class='kw'>sep.out</span> <span class='kw'>=</span> <span class='st'>""</span>)
<span class='no'>loc2loc.nosep</span>


<span class='co'>## Format tool</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)

<span class='co'>### Adding zeros</span>

<span class='no'>example</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/as.matrix'>as.matrix</a></span>(<span class='no'>genotype</span>[<span class='fl'>1</span>:<span class='fl'>10</span>,])
<span class='fu'>mode</span>(<span class='no'>example</span>) <span class='kw'>&lt;-</span> <span class='st'>"character"</span>
<span class='co'>### example data</span>
<span class='no'>example</span>
<span class='no'>recoded</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.format.html'>eco.format</a></span>(<span class='no'>example</span>, <span class='kw'>ncod</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>nout</span> <span class='kw'>=</span> <span class='fl'>3</span>)
<span class='co'>### recoded data</span>
<span class='no'>recoded</span>
<span class='co'>### leading zeros</span>
<span class='no'>recoded2</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.format.html'>eco.format</a></span>(<span class='no'>example</span>, <span class='kw'>ncod</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>nout</span> <span class='kw'>=</span> <span class='fl'>3</span>,
<span class='kw'>fill.mode</span> <span class='kw'>=</span> <span class='st'>"first"</span>)
<span class='co'>### recoded data</span>
<span class='no'>recoded2</span>


<span class='co'>### Tetraploid data, separating alleles with a "/"</span>
<span class='no'>tetrap</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/as.matrix'>as.matrix</a></span>(<span class='no'>example</span>)
<span class='co'>### simulated tetraploid example data</span>
<span class='no'>tetrap</span> <span class='kw'>&lt;-</span> <span class='fu'>matrix</span>(<span class='fu'>paste</span>(<span class='no'>example</span>,<span class='no'>example</span>, <span class='kw'>sep</span> <span class='kw'>=</span> <span class='st'>""</span>), <span class='kw'>ncol</span> <span class='kw'>=</span> <span class='fu'><a href='http://www.rdocumentation.org/packages/raster/topics/ncell'>ncol</a></span>(<span class='no'>example</span>))
<span class='no'>recoded</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='eco.format.html'>eco.format</a></span>(<span class='no'>tetrap</span>, <span class='kw'>ncod</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>ploidy</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>sep.out</span> <span class='kw'>=</span> <span class='st'>"/"</span>)
<span class='co'>### recoded data</span>
<span class='no'>recoded</span>


<span class='co'>## Creating input data for Geneland with an ecogen object</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>eco.2geneland</span>(<span class='no'>eco</span>, <span class='fl'>1</span>)


<span class='co'>## Exporting an ecogen genetic data frame into Genepop format</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>eco.2genepop</span>(<span class='no'>eco</span>, <span class='kw'>grp</span> <span class='kw'>=</span> <span class='st'>"pop"</span>, <span class='kw'>name</span> <span class='kw'>=</span> <span class='st'>"infile.genepop.txt"</span>)
<span class='co'>### an output file "infile.genepop.txt" is generated in the working directory</span>


<span class='co'>## Converting a diploid ecogen genetic data frame into a gstudio object</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>gsteco</span> <span class='kw'>&lt;-</span> <span class='fu'>eco.2gstudio</span>(<span class='no'>eco</span>, <span class='st'>"separated"</span>)
<span class='no'>gsteco</span>


<span class='co'>## Converting an ecogen genetic data frame into a hierfstat data frame</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='no'>hiereco</span> <span class='kw'>&lt;-</span> <span class='fu'>eco.2hierfstat</span>(<span class='no'>eco</span>, <span class='st'>"pop"</span>)


<span class='co'>## Exporting an ecogen genetic data frame into SPAGeDI format</span>

<span class='fu'>data</span>(<span class='no'>eco.test</span>)
<span class='fu'>eco.2spagedi</span>(<span class='no'>eco</span>, <span class='st'>"pop"</span>, <span class='kw'>ndig</span> <span class='kw'>=</span> <span class='fl'>1</span>,<span class='kw'>int</span><span class='kw'>=</span><span class='fl'>2</span>, <span class='kw'>smax</span><span class='kw'>=</span><span class='fl'>6</span>, <span class='kw'>name</span><span class='kw'>=</span><span class='st'>"infile.spagedi.txt"</span>)

<span class='co'># }</span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    

 Leandro Roser, Juan Vilardi, Beatriz Saidman and Laura Ferreyra
 
 Maintainer: Leandro Roser &lt;leandroroser@ege.fcen.uba.ar&gt;

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Leandro Roser, Juan Vilardi, Beatriz Saidman, Laura Ferreyra.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
